src/utils/getFilteredPeople.ts

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { useMemo } from 'react';
+import { Person } from '../types/Person';
+import { SortType } from '../types/SortType';
+
+export const getFilteredPeople = (
+  people: Person[],
+  query: string,
+  sex: string,
+  numbers: string[],
+  sort: string,
+  order: string,
+): Person[] => {
+  let preparedPeople = useMemo(() => {
+    return people.map(person => {
+      const father = people.find(human => human.name === person.fatherName);
+
+      const mother = people.find(human => human.name === person.motherName);
+
+      return { ...person, father, mother };
+    });
+  }, [people]);
+
+  switch (sort) {
+    case SortType.Name: {
+      preparedPeople = preparedPeople.sort((person1, person2) =>
+        person1.name.localeCompare(person2.name),
+      );
+
+      break;
+    }
+
+    case SortType.Sex: {
+      preparedPeople = preparedPeople.sort((person1, person2) =>
+        person1.sex.localeCompare(person2.sex),
+      );
+
+      break;
+    }
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Sex:
      preparedPeople = preparedPeople.sort((person1, person2) =>
        person1[sort].localeCompare(person2[sort]),
      );

      break;
```
we can group some cases